### DISCUSSION TITLE
Add support for BTTV emote modifiers

### DISCUSSION THREAD

**Discussion by NBruderman** [Chatterino/chatterino2#4470] [Ideas]
Recently, Bttv implemented some modifiers, that when used, modify the emote according to this list:

w! makes the emote wide
v! flips it vertically
h! flips it horizontally
z! between two emotes makes it so that there is no space

the feature is free to use. and widely being used in some chats

relevant commit from bttv
[https://github.com/night/betterttv/commit/b60d02091f2565b80ff36bc72ca1014a76df69f2](https://github.com/night/betterttv/commit/b60d02091f2565b80ff36bc72ca1014a76df69f2)

**Comment by Felanbird** [comment]
We've had many discussions about this, and the summary has been "we'll see".

In comparison to `_HF` and `_SQ` Twitch emotes, which send Chatterino a brand new emote and we just display it, the BTTV modifiers are doing all the handling in browser, meaning we will have to do the heavy lifting to implement them into Chatterino, as BTTV just sends us a normal emote.

**Comment by Pafon22** [comment]
c! to Black and White emote too.

**Reply by Nerixyz** [reply]
~~This is a bit more complex than the other ones, since it modifies the actual pixels of the image and not the layout. Maybe specifying a composition mode with some black/white rectangle below could be fine?~~

There's actually a private [`QPixmapFilter`](https://github.com/qt/qtbase/blob/77a4ff4036330ea445b0dab8132cf143533106a7/src/widgets/effects/qpixmapfilter_p.h#L32-L57) class. [`QPixmapColorizeFilter`](https://github.com/qt/qtbase/blob/77a4ff4036330ea445b0dab8132cf143533106a7/src/widgets/effects/qpixmapfilter_p.h#L105-L121) is a subclass and seems to do what you want.

There's a disclaimer, but this could be extracted (like [`QColorPicker`](https://github.com/Chatterino/chatterino2/blob/76527073cfe5e65d5fcae01cdd3e98227a5298c1/src/widgets/helper/QColorPicker.hpp#L46-L53)):

>This file is not part of the Qt API.  It exists purely as an implementation detail.  This header file may change from version to version without notice, or even be removed.
>
> We mean it.
